src/coreComponents/constitutive/solid/ModifiedCamClay.hpp

```diff
@@ -0,0 +1,568 @@
+                      real64 & df_dp,
+                      real64 & df_dq,
+                      real64 & df_dpc,
+                      real64 & df_dp_dve,
+                      real64 & df_dq_dse ) const;
+
+
+
+  GEOSX_HOST_DEVICE
+  virtual void smallStrainUpdate( localIndex const k,
+                                  localIndex const q,
+                                  real64 const ( &strainIncrement )[6],
+                                  real64 ( &stress )[6],
+                                  real64 ( &stiffness )[6][6] ) const override final;
+
+  GEOSX_HOST_DEVICE
+  virtual void smallStrainUpdate( localIndex const k,
+                                  localIndex const q,
+                                  real64 const ( &strainIncrement )[6],
+                                  real64 ( &stress )[6],
+                                  DiscretizationOps & stiffness ) const final;
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  virtual void saveConvergedState( localIndex const k,
+                                   localIndex const q ) const override final
+  {
+    ElasticIsotropicPressureDependentUpdates::saveConvergedState( k, q );
+    m_oldPreConsolidationPressure[k][q] = m_newPreConsolidationPressure[k][q];
+  }
+
+private:
+
+  /// A reference to the ArrayView holding the virgin compression index for each element.
+  arrayView1d< real64 const > const m_virginCompressionIndex;
+
+  /// A reference to the ArrayView holding the slope of the critical state line for each element.
+  arrayView1d< real64 const > const m_cslSlope;
+
+  /// A reference to the ArrayView holding the new preconsolidation pressure for each integration point
+  arrayView2d< real64 > const m_newPreConsolidationPressure;
+
+  /// A reference to the ArrayView holding the old preconsolidation presure for each integration point
+  arrayView2d< real64 > const m_oldPreConsolidationPressure;
+
+};
+
+
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void ModifiedCamClayUpdates::evaluateYield( real64 const p,
+                                            real64 const q,
+                                            real64 const pc,
+                                            real64 const M,
+                                            real64 const Cc,
+                                            real64 const Cr,
+                                            real64 const bulkModulus,
+                                            real64 const mu,
+                                            real64 & f,
+                                            real64 & df_dp,
+                                            real64 & df_dq,
+                                            real64 & df_dpc,
+                                            real64 & df_dp_dve,
+                                            real64 & df_dq_dse ) const
+{
+
+  df_dp = -pc + 2. * p;
+  df_dq = 2. * q /(M*M);
+  df_dpc = -p;
+  real64 dpc_dve = -1./(Cc-Cr) * pc;
+  df_dp_dve = 2. * bulkModulus + dpc_dve;
```

Comment:
```suggestion
  df_dp_dve = 2. * bulkModulus - dpc_dve;
```

Reply:
I think the equation is already correct. This will affect the convergence if edited.